File: sdks/dotnet/src/Mobiscroll.Connect/Exceptions/ServerException.cs
namespace Mobiscroll.Connect.Exceptions;

public sealed class ServerException : MobiscrollConnectException
{
    public int StatusCode { get; }

    public ServerException(string message, int statusCode)
        : base(message, "SERVER_ERROR")
    {
        StatusCode = statusCode;
    }
}

File: sdks/java/connect-sdk/src/main/java/com/mobiscroll/connect/exceptions/ServerException.java
package com.mobiscroll.connect.exceptions;

/** Thrown on HTTP 5xx — the Mobiscroll Connect service or an upstream calendar provider failed. */
public class ServerException extends MobiscrollConnectException {

    private static final long serialVersionUID = 1L;

    private final int statusCode;

    public ServerException(String message, int statusCode) {
        super(message);
        this.statusCode = statusCode;
    }

    public int getStatusCode() {
        return statusCode;
    }
}
